src/pim-types.ts

```diff
@@ -181,6 +183,15 @@ export class TaskType extends EventType {
     return this.component.getFirstPropertyValue('priority');
   }
 
+  set tags(tags: string[]) {
+    this.component.updatePropertyWithValue('categories', tags.join(','));
```

Comment:
Yeah after I made [this comment](https://github.com/etesync/etesync-web/pull/74#discussion_r392524201) I did some more investigating and realized that the issue with the library isn't `setValues` but `getValues`. It _is_ able to correctly create valid ICAL objects. But it behaves a bit strangely when retrieving values. If there is an empty array value it will return an array with an empty string: `[""]`.

I figure doing it this way is just easier. Also, `updatePropertyWithValue` is more convenient than `getFirstProperty.setValues` because it will automatically create the property if it doesn't exist.

Reply:
I wonder if we are abusing ical.js here... :)
No worries, as if you are abusing it here, we are also abusing it elsewhere. I just wonder what's going on there under the hood.

Reply:
Haha I think we are. Seemingly the "correct" way to handle list values is to use `setValues`/`getValues`, but this is broken in the library so 🤷🏼‍♀️

I tried looking under the hood, but the codebase was so complicated. Perhaps I can raise an issue and hopefully it will be fixed down the line (or I can make a PR when I find some time).

For now, I am fine with our hacky method.

Reply:
My hunch is that somewhere they are `split`ing a comma delimited string into an array. But when you `split` an empty string it actually gives you an array with a single empty string in it.
```js
"".split(',') // -> [""]
```

Reply:
If memory serves I already opened a ticket there. And yeah, the code is very complex and I wouldn't even know if our changes are correct or not. iCal works in mysterious ways.

Reply:
The split thing: ah, quite possibly! So maybe you can open a PR there if you fancy. If you ever end up doing it, CC me, so I know to fix the rest of the code once ical.js is fixed.